src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -503,6 +503,8 @@ void prepare_buffer_fusing::run(program& p) {
                     return;
                 if (user->is_type<loop>() || user->is_type<non_max_suppression>())
                     return;
+            }
+            for (auto user : node.get_users()) {
```

Comment:
This is to fix 131544 issue.
For the case which crop has 2 children from LSTMSequence, concat and reshape, this is not to reach at can_reshape_be_optimized() check which may cause exception issue because of invalidated reshape's output layout. Other fix could be add "reshape->get_output_layout()" before can_reshape_be_optimized() but, I think this fix is simpler.

Reply:
That sounds weird because at buffer fusing stage, all nodes should have valid layout. Why the reshape's output is invalidated..?

Reply:
it's invalidated during prepare_padding and never validated until the the check in prepare buffer fusing. "node.get_output_layout" which can validate the nodes is followed after can_reshape_be_optimized() check.

Some months ago, can_reshape_be_optimized() doesn't require output_layout and this doesn't cause problem.

Reply:
Having not validated node at this moment is problem (a potential probelm) , but this change itself is okay.